include/arch/arm/cortex_m/mpu/arm_mpu.h

```diff
@@ -102,18 +102,18 @@ struct arm_mpu {
 
 /* Some helper defines for common regions */
 #define REGION_USER_RAM_ATTR(size) \
-		(NORMAL_OUTER_INNER_NON_CACHEABLE_NON_SHAREABLE | \
+		(NORMAL_OUTER_INNER_WRITE_BACK_NON_SHAREABLE | \
 		 NOT_EXEC | size | FULL_ACCESS)
 #define REGION_RAM_ATTR(size) \
-		(NORMAL_OUTER_INNER_NON_CACHEABLE_NON_SHAREABLE | \
+		(NORMAL_OUTER_INNER_WRITE_BACK_NON_SHAREABLE | \
```

Comment:
Are we sure that write back on the flash makes sense?

Reply:
This is still a RAM attribute macro, btw. The flash macros are below.

Reply:
@galak to (try to) elaborate on this; does it make more sense to use Write-Through?
Or you wihder whether cachability on Flash actually makes sense?

Reply:
If we enable cache on RAM, we will have to introduce some code to maintain cache coherency on multi-core devices as well as update all drivers using DMA on SoCs with data cache.

Reply:
> If we enable cache on RAM, we will have to introduce some code to maintain cache coherency on multi-core devices as well as update all drivers using DMA on SoCs with data cache (however I am not sure if there are such devices in the Zephyr world).

This alone won't enable the cache on RAM, it rather allows to disable the cache activation when `SCB_EnableICache()` and/or `SCB_EnableDCache()` have been called. If the MPU is disabled, the cache might already be enabled and coherency already has to be maintained. For example the NXP i.MX RT family enables the cache by default.

I agree that cache coherency should be maintained if cache is enabled, but that's something to check when adding code to enable the cache on a given SoC (like in #8647), not at the MPU level.

Reply:
The _ARM v7-M Architecture Reference Manual_, in section B3.1, clearly says that the code area (0x00000000-
0x1FFFFFFF) can only be configure as write-through. So I believe `NORMAL_OUTER_INNER_WRITE_THROUGH_NON_SHAREABLE` should be used instead.

Reply:
@aurel32 I agree, that's what I also proposed.